internal/service/cluster/resource_cluster.go

```diff
@@ -803,15 +810,21 @@ func resourceMongoDBAtlasClusterUpdate(ctx context.Context, d *schema.ResourceDa
 		}
 	}
 
+	replicationSpecs, err := expandReplicationSpecs(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(errorClusterUpdate, clusterName, err))
+	}
 	if d.HasChange("replication_specs") {
-		replicationSpecs, err := expandReplicationSpecs(d)
-		if err != nil {
-			return diag.FromErr(fmt.Errorf(errorClusterUpdate, clusterName, err))
-		}
-
 		cluster.ReplicationSpecs = replicationSpecs
 	}
 
+	err = validateProviderRegionName(d.Get("cluster_type").(string), d.Get("provider_region_name").(string), replicationSpecs)
+	if err != nil {
+		if cluster.ProviderSettings != nil {
+			cluster.ProviderSettings.RegionName = ""
```

Comment:
it feels like here we are swallowing an error and going ahead by resetting the field. If that's the right interpretation, should we just return an error?

Reply:
We are swallowing the error for backwards compatibility here, so that if any existing users have the `provider_region_name` configured for a multi-region cluster (or updating a single region to multi-region) then they don't get any errors on running updates. But I guess this could lead to non-empty plans later if user has `provider_region_name` configured.

Let me cross check and will update this to return error.

Reply:
Looked into this, we need to swallow the error during update. I have added a test where a single region cluster with this param specified is updated to a multi-region one, in such cases (or existing cases where users with multi-region clusters might have this param, if specified, in their state)

This is what happens in such cases:
1. a single-region cluster is created with `provider_region_name` set. User's state also has this param set.
2. user updates config to multi-region and adds more region-configs or replication_specs and removes `provider_region_name` from config and UPDATE handler is invoked.
3. Since `provider_region_name` is Optional & Computed, TF still reads the old value of this param from the state and user will get an error even though they have removed the param from their config.

To handle cases like above, we ignore `provider_region_name` param for multi-region clusters during update.

Please let me know if there's a better way to go about this.

Reply:
Thanks for the details. Just to be sure, is `d.HasChange("provider_region_name")`/`d.GetChange("provider_region_name")` capable of detecting that the user has removed the attribute?

Reply:
No, verified that `d.HasChange("provider_region_name")/d.GetChange("provider_region_name")` don't detect that user has removed this attribute:

`old, new := d.GetChange("provider_region_name")`
log output:
```
2024/01/12 15:30:58 [INFO] old prn: US_EAST_1
2024/01/12 15:31:01 [INFO] new prn: US_EAST_1
```

Reply:
I see, I believe we don't have another option. If we go this way we can create a ticket to incorporate the error message in the update operation, and link to the terraform framework migration epic. In that case we will be able to easily identify what is defined in the configuration vs the state.

Sorry for another question, if we don't reset the field how unfriendly is the error message from the Atlas API? Just thinking if leaving that error is better than modifying the value of "provider_region_name" and generating inconsistency in the state with what the user has defined in his configuration.

Reply:
>currently a user is not able to transition from single region cluster with provider_region_name to a multi-region right? 
right, they could potentially run into issues like https://jira.mongodb.org/browse/HELP-51429 

>If we go this way we can create a ticket to incorporate the error message in the update operation and link to the terraform framework migration epic.
Since this is cluster resource we don't have plans to migrate this to the Framework